plugins/modules/system/sapcontrol.py

```diff
@@ -0,0 +1,389 @@
+        - RestartServiceWait
+        - WaitforServiceStarted
+        - CheckHostAgent
+        type: str
+    parameter:
+        description:
+            - The parameter to pass to the function.
+        required: false
+        type: str
+    force:
+        description:
+            - Forces the execution of the function C(Stop).
+        required: false
+        default: false
+        type: bool
+author:
+    - Rainer Leber (@RainerLeber)
+    - Robert Kraemer (@rkpobe)
+notes:
+    - Does not support C(check_mode).
+'''
+
+EXAMPLES = r"""
+- name: GetProcessList with sysnr
+  community.sap_libs.sapcontrol:
+    hostname: 192.168.8.15
+    sysnr: 01
+    function: GetProcessList
+
+- name: GetProcessList with custom port
+  community.sap_libs.sapcontrol:
+    hostname: 192.168.8.15
+    function: GetProcessList
+    port: "50113"
+
+- name: ParameterValue
+  community.sap_libs.sapcontrol:
+    hostname: 192.168.8.15
+    sysnr: 01
+    username: hdbadm
+    password: test1234#
+    function: ParameterValue
+    parameter: ztta
+"""
+
+RETURN = r'''
+msg:
+    description: Success-message with functionname.
+    type: str
+    returned: always
+    sample: 'Succesful execution of: GetProcessList'
+out:
+    description: The full output of the required function.
+    type: list
+    elements: dict
+    returned: always
+    sample: [{
+            "item": [
+                {
+                    "description": "MessageServer",
+                    "dispstatus": "SAPControl-GREEN",
+                    "elapsedtime": "412:30:50",
+                    "name": "msg_server",
+                    "pid": 70643,
+                    "starttime": "2022 03 13 15:22:42",
+                    "textstatus": "Running"
+                },
+                {
+                    "description": "EnqueueServer",
+                    "dispstatus": "SAPControl-GREEN",
+                    "elapsedtime": "412:30:50",
+                    "name": "enserver",
+                    "pid": 70644,
+                    "starttime": "2022 03 13 15:22:42",
+                    "textstatus": "Running"
+                },
+                {
+                    "description": "Gateway",
+                    "dispstatus": "SAPControl-GREEN",
+                    "elapsedtime": "412:30:50",
+                    "name": "gwrd",
+                    "pid": 70645,
+                    "starttime": "2022 03 13 15:22:42",
+                    "textstatus": "Running"
+                }
+                ]
+            }]
+'''
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+import traceback
+try:
+    from suds.client import Client
+    from suds.sudsobject import asdict
+except ImportError:
+    HAS_SUDS_LIBRARY = False
+    SUDS_LIBRARY_IMPORT_ERROR = traceback.format_exc()
+else:
+    HAS_SUDS_LIBRARY = True
+
+
+def choices():
+    retlist = ["Start", "Stop", "RestartInstance", "Shutdown", "InstanceStart", 'GetProcessList',
```

Comment:
ok. this is a little bit nitpicking but "Start","Stop" and later 'GetProcessList'.  maybe uniformly. (i think that was my fault, since I started the list 🤔

Reply:
I think it won't matter

Reply:
Yeah. Its just for aestethics ;-)